pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -151,20 +152,29 @@ func add(mgr manager.Manager, r *MemberRollReconciler) error {
 func toRequests(ctx context.Context, cl client.Client, namespace string) []reconcile.Request {
 	log := common.LogFromContext(ctx)
 	list := &maistrav1.ServiceMeshMemberRollList{}
-	err := cl.List(ctx, list, client.MatchingFields{"spec.members": namespace})
+	err := cl.List(ctx, list)
 	if err != nil {
 		log.Error(err, "Could not list ServiceMeshMemberRolls")
 	}
 
 	var requests []reconcile.Request
 	for _, smmr := range list.Items {
-		requests = append(requests, reconcile.Request{
-			NamespacedName: common.ToNamespacedName(&smmr),
-		})
+		if smmr.Spec.IsClusterScoped() || contains(smmr.Spec.Members, namespace) {
+			requests = append(requests, reconcile.Request{
+				NamespacedName: common.ToNamespacedName(&smmr),
+			})
+		}
 	}
 	return requests
 }
 
+func contains(members []string, ns string) bool {
+	for _, member := range members {
+		return member == ns
```

Comment:
Is this right? It always returns on first interaction

Reply:
oh, wow, what was I smoking here?

I've now fixed this.